### DISCUSSION TITLE
Docker digest updating & Compose build image tag

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#12039] [Request Help]
We currently use a `docker-compose.yml` like the following to maintain our `app` service for development:

```yaml
services:
  app:
    build:
      context: .docker/app
    image: <vendor>/<project>-app:v<version>
```

Here a `.docker/app/Dockerfile` is used to build the `app` image which is then named and tagged as `<vendor>/<project>-app:v<version>` (e.g. `acme/foo-app:v5`). Whenever there is a change to the `Dockerfile` the service `image` tag is bumped manually which ensures that other developers as well as our CI picks up the changes automatically on `docker-compose up -d` / `docker-compose run`.

We use base images in our `Dockerfile` which are rebuilt frequently and the tags are updated accordingly. Thus the same image tag will point to different digests over time making for a rather unreliable result. Depending on when and where the `app` image is built, the result can be completely different.

Given the current issues with the expiry of the Let's Encrypt certificate DST Root CA X3 we thus started thinking about Docker image [digest pinning](https://docs.renovatebot.com/docker/#digest-pinning) and updating as supported by Renovate.

However, even if Renovate would suggest digest updates for our `Dockerfile` these updates would only have an effect if developers and our CI actively rebuild the `app` service. The necessary service `image` tag bump is missing here to ensure a rebuild.

Long story short: are there best practices/ideas how to handle this within Renovate?

I already had a look at the [`regex` manager](https://docs.renovatebot.com/modules/manager/regex/) but don't know if Renovate could bump the `image` tag on its own given that there is no data source for custom build images. Also we'd need a single update which combines the digest update in the `Dockerfile` and the service `image` tag bump

**Comment by rarkins** [comment]
> there is no data source for custom build images.

Can you explain this more? You mentioned earlier it's "named and tagged" but do you mean it's not actually *pushed* to a registry?

**Reply by mbrodala** [reply]
This is about the `image` in the `docker-compose.yml`. These only exist by name and locally, they are not pushed to a registry. So basically every developer and CI pipeline will have their own version.

Also I'm not sure pushing them would help here since in the end we'd need a single PR which contains the base image digest update as well as an update to `image` in the `docker-compose.yml`.
